Authentication API #Issue_Number-1

Sureshwebdeveloper opened the pull request:
I have Writed user Authentication Api 

## Flow Of My Cod
1. User Register New Account --> email, password,confirm password  ---> Sending Token Access Token,Authentication Token

2. User Create Our Profile Via Registered Email And Update A Other Deatiles
      name,
      gender,
      age,
      country,
      phoneNumber,
      countryCode,
      description,
      social media link,
      profile picture, 

3. Login User Via Email Password Or Registered PhoneNumber And Return Only Authentication Token

If I need to make any other changes, please provide them. I am excited to work on that.
@Mustafiz04 

this close #1

Mustafiz04 commented on api/controller/auth.js:
```diff
@@ -19,6 +20,21 @@ const generateToken = (user) => {
   }
 };
 
+// Access Token 1hr
+const generateAccessToken = (access) => {
+  const payload = {
+    exp:
+      Math.floor(Date.now() / 1000) + parseInt(process.env.ACCESS_TOKEN_EXPIRY),
+    data: access,
+  };
+  try {
+    return jwt.sign(payload, secret);
+  } catch (error) {
+    console.error("[Error] Failed to generate Access token:", error);
+    throw error;
+  }
+};
+
```
Also can you please explain about this function? Difference between existing `generateToken` ??

Sureshwebdeveloper replied:
nothing to be on big it will be add a time of different like 1hour, 2hour different times,  now i will update code  parameter now we can get a 2 parameter first one for data, 2nd one for time.

Mustafiz04 commented on api/controller/auth.js:
```diff
@@ -61,21 +96,106 @@ const signup = async (req, res) => {
+      return res.status(400).json({ message: "Please Enter Email" });
+    }
+
+    if (
+      (!name,
+      !gender,
+      !age,
+      !country,
+      !phoneNumber,
+      !countryCode,
+      !description,
+      !socialMediaLink,
+      !profilePicture)
+    ) {
+      return res.status(400).json({ message: "Please Fill All Fields" });
+    }
```
Can we do this validation using express-validator middleware ?? Same for all validations?
https://express-validator.github.io/docs/

Sureshwebdeveloper replied:
ok i will add express-validator validations

Sureshwebdeveloper commented on api/routes/auth.js:
```diff
@@ -1,9 +1,17 @@
 
-AuthRouter.post("/signup", signup);
 AuthRouter.post("/signin", signin);
+AuthRouter.post("/profile", accessToken, validateToken, createProfile);
```
ok i will remove a middleware either one only add.......

Sureshwebdeveloper commented on api/middleware/auth.js:
```diff
@@ -25,6 +25,27 @@ export const validateToken = async (req, res, next) => {
   }
 };
 
+export const accessToken = async (req, res, next) => {
+  try {
+    const token = req.headers["accesstoken"] || req.headers["accessToken"];
+
+    if (!token) {
+      return res
+        .status(401)
+        .json({ message: "No token provided accesstoken", status: 401 });
+    }
+
+    const decoded = jwt.verify(token, secret);
+    if (decoded.exp < Math.floor(Date.now() / 1000)) {
+      return res.status(401).json({ message: "Token expired", status: 401 });
+    }
+    req.access = decoded.data;
+    next();
+  } catch (error) {
+    return res.status(401).json({ message: "Token is expried." });
+  }
+};
+
```
i thought we need to validate a two token at a time so i wrote a separate function for validation..

Sureshwebdeveloper commented:
@Mustafiz04  Please Check For Now I Have Update A Latest Commit and I have Removed a unnecessary token Validation and  Token Generation , added a Express Validator
